roiextractors/extractors/bioformatsimagingextractors/bioformatsimagingbaseextractor.py

```diff
@@ -57,31 +58,20 @@ def _start_javabridge_vm(self):
     def _kill_javabridge_vm(self):
         javabridge.kill_vm()
 
-    def get_frame(self, frame_idx, channel=0):
-        assert frame_idx < self.get_num_frames()
-        plane = self._reader.read(t=frame_idx).T
-        return plane
-
+    @check_get_frames_args
     def get_frames(self, frame_idxs, channel=0):
-        frame_idxs = np.array(frame_idxs)
-        assert np.all(frame_idxs < self.get_num_frames())
         planes = np.zeros((len(frame_idxs), self._size_x, self._size_y))
         for i, frame_idx in enumerate(frame_idxs):
             plane = self._reader.read(t=frame_idx).T
             planes[i] = plane
         return planes
 
+    # TODO make this memmap if too large
+    @check_get_videos_args
     def get_video(self, start_frame=None, end_frame=None, channel=0):
-        if start_frame is None:
```

Comment:
Is there a reason to keep `get_video`? Is there a use-case that can't be handled by `get_frames`?

Reply:
You can do everything with `get_frames`, but personally I'd like to keep both as `get_video` is more compact for continuous data. If you prefer to remove I can do it.

Reply:
OK, I'm just thinking about trying to keep the API minimal and consistent. How about we define `get_video` in `ImagingExtractor` as `return self.get_frames(range(start_frame, end_frame), channel=channel)` and overwrite as necessary? In this case, I think they would do the same thing, so I don't think overwriting would be necessary.

Reply:
That's a great idea :)

Performance-wise, slicing should be better, but we can use the decorator of the get_frames to make a slice in case frame_idxs are continuous. 

Are you ok in keeping the decorators for now, or should I just remove them and make the function?

Reply:
Yes, let's leverage a continuous slice where we can for performance, but as you said this can be internal to `get_frames`. I'm not sure if I'd want to do it with a decorator though.